### ISSUE TITLE
Lutris wines can't open browser links with KIO Client

### ISSUE THREAD

**Issue by Yeshey** [NixOS/nixpkgs#235995]
### Describe the bug
Whenever an application running through a Lutris wine version tries to open a link, I get the following window:
![image](https://github.com/NixOS/nixpkgs/assets/41551785/03864c1e-538b-44ba-a5b4-30effbfc82c1)
But when I press continue, the link is not open.

This can also be reproduced by on Lutris doing "Open Bash terminal" and running `wine winebrowser google.com`

### Steps To Reproduce
1. Run lutris. (`NIXPKGS_ALLOW_UNFREE=1 nix-shell -p lutris --run lutris`)
2. Go to wine > Manage versions and install a wine version if none installed already.
3. Go to main menu or select a game that runs with wine and to the right of the play button press the wine icon > open Bash terminal
4. In the new terminal run `wine winebrowser google.com`

### Expected behavior
Should open links in my browser of choice.

### Additional context

The error I get in the logs is this:
```
kf.config.core: Access to ' "/etc/profiles/per-user/yeshey/share/applications/vivaldi-stable.desktop" ' denied, not owned by root and executable flag not set.
kf.kio.core: Unable to open replacement file for "/etc/profiles/per-user/yeshey/share/applications/vivaldi-stable.desktop" "Existing file /etc/profiles/per-user/yeshey/share/applications/vivaldi-stable.desktop is not writable"
```

That file seems to be owned by root tho:
```bash
> ll /etc/profiles/per-user/yeshey/share/applications/vivaldi-stable.desktop
lrwxrwxrwx 2 root root 103 jan  1  1970 /etc/profiles/per-user/yeshey/share/applications/vivaldi-stable.desktop -> /nix/store/04vj57an335hw7wclgd33c93mikk8k58-home-manager-path/share/applications/vivaldi-stable.desktop
```

This might not be specific to lutris, or might be a common problem, I noticed the application `heroic` has this problem too, for example by opening heroic > Stores > Epic Store > Open in browser

([Also reported on the wine reddit](https://www.reddit.com/r/wine_gaming/comments/xuic32/browser_links_not_working/))

### Notify maintainers

@Madouura 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.108, NixOS, 22.11 (Raccoon), 22.11.20230428.7449971`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixos-22.11, nixpkgs-22.11"`
 - channels(yeshey): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

```console
> nix-shell -p neofetch --run "neofetch"
          ▗▄▄▄       ▗▄▄▄▄    ▄▄▄▖            yeshey@nixOS-Laptop 
          ▜███▙       ▜███▙  ▟███▛            ------------------- 
           ▜███▙       ▜███▙▟███▛             OS: NixOS 22.11.20230428.7449971 (Raccoon) x86_64 
            ▜███▙       ▜██████▛              Host: LENOVO LNVNB161216 
     ▟█████████████████▙ ▜████▛     ▟▙        Kernel: 5.15.108 
    ▟███████████████████▙ ▜███▙    ▟██▙       Uptime: 4 hours, 36 mins 
           ▄▄▄▄▖           ▜███▙  ▟███▛       Packages: 2114 (nix-system), 1930 (nix-user), 29 (flatpak) 
          ▟███▛             ▜██▛ ▟███▛        Shell: bash 5.1.16 
         ▟███▛               ▜▛ ▟███▛         Resolution: 1920x1080 
▟███████████▛                  ▟██████████▙   DE: Plasma 
▜██████████▛                  ▟███████████▛   WM: KWin 
      ▟███▛ ▟▙               ▟███▛            Icons: breeze-dark [GTK2/3] 
     ▟███▛ ▟██▙             ▟███▛             Terminal: .konsole-wrappe 
    ▟███▛  ▜███▙           ▝▀▀▀▀              CPU: Intel i7-9750H (12) @ 4.500GHz 
    ▜██▛    ▜███▙ ▜██████████████████▛        GPU: NVIDIA GeForce RTX 2060 Mobile 
     ▜▛     ▟████▙ ▜████████████████▛         GPU: Intel CoffeeLake-H GT2 [UHD Graphics 630] 
           ▟██████▙       ▜███▙               Memory: 9081MiB / 15878MiB 
          ▟███▛▜███▙       ▜███▙
         ▟███▛  ▜███▙       ▜███▙                                     
         ▝▀▀▀    ▀▀▀▀▘       ▀▀▀▘                                     

```

**Comment by DaKingof** [comment]
I get this everywhere. Any app that is sandboxed pretty much is blocked. VS Code links don't work. Heroic Launcher I can't log into games from. It's a mess. This issue is seriously bad and seems pretty much ignored. I went into chat asking for help with it and was ignored there on Element. I don't know what else to do. This is so frustrating.

**Comment by LDprg** [comment]
Also have this error! Pls fix!

**Comment by LDprg** [comment]
It seem like this error foes not accur with firefox as default browser. I only got it when brave was my default browser.

**Comment by GuillaumeDesforges** [comment]
This error also happens in FHS VSCode, stopping it from opening the browser.

**Comment by tomsjansons** [comment]
I was suffering from the same problem with Brave and took me ages to figure this one out but there is a workaround - the problem is caused by perms on the .dekstop file.
TLDR: copy your desktop file to `~/.local/share/applications`, this should have the right perms and the URLs will start to work

Long explanation - I noticed that Firefox worked, but Brave did not. Looking at `journalctl -f` I saw errors about permissions
![logs](https://github.com/NixOS/nixpkgs/assets/1428917/7cea0a46-f644-41ac-b589-2c368ffbc2d2)
Depending on if Brave is installed for user or system wide, the location will differ. Find the location, follow the symlink and find the desktop files in the nix store
I compared Firefox and Brave desktop files to see that they indeed had different perms
![firefox](https://github.com/NixOS/nixpkgs/assets/1428917/56d57742-e0c2-434f-99f5-6acbb7e43f7b)
![brave](https://github.com/NixOS/nixpkgs/assets/1428917/a935fd8e-0b3f-4c87-9df8-32fb0774955d)
you can now copy your Brave/Vivaldi desktop file to your home, adjust perms as necessary (I didn't need to) and fingers crossed it works!

**Comment by sammypanda** [comment]
> I was suffering from the same problem with Brave and took me ages to figure this one out but there is a workaround - the problem is caused by perms on the .dekstop file. TLDR: copy your desktop file to `~/.local/share/applications`, this should have the right perms and the URLs will start to work
> 
> Long explanation - I noticed that Firefox worked, but Brave did not. Looking at `journalctl -f` I saw errors about permissions ![logs](https://private-user-images.githubusercontent.com/1428917/298195033-7cea0a46-f644-41ac-b589-2c368ffbc2d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1MDMwMTQsIm5iZiI6MTcxMTUwMjcxNCwicGF0aCI6Ii8xNDI4OTE3LzI5ODE5NTAzMy03Y2VhMGE0Ni1mNjQ0LTQxYWMtYjU4OS0yYzM2OGZmYmMyZDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjdUMDEyNTE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmI3ZTA1OWQ4MTk2OTZmZTRmY2Y0NTI1ZWNlZWE1M2U4NzQ5MTM1OTgzMzc1NTk5ZTdiYzBlNGZmZTFiMGU5MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6gFj8e_uFs9UwaPLIhRWlpJeghrny97mceMPF71J81A) Depending on if Brave is installed for user or system wide, the location will differ. Find the location, follow the symlink and find the desktop files in the nix store I compared Firefox and Brave desktop files to see that they indeed had different perms ![firefox](https://private-user-images.githubusercontent.com/1428917/298195117-56d57742-e0c2-434f-99f5-6acbb7e43f7b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1MDMwMTQsIm5iZiI6MTcxMTUwMjcxNCwicGF0aCI6Ii8xNDI4OTE3LzI5ODE5NTExNy01NmQ1Nzc0Mi1lMGMyLTQzNGYtOTlmNS02YWNiYjdlNDNmN2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjdUMDEyNTE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjQ3MjczOWY4ODBlZjk1Mzk0Y2QxYmEzMTRhZTU3Nzc4ZWZhNzA0ZmNlYjIyMzhkOTgyMzU5MjBiY2MwMDJiNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.4zMxGVfQQzDzcnsdwNxz0GxCRO3_8Q3xO5H2FMB3q8s) ![brave](https://private-user-images.githubusercontent.com/1428917/298195131-a935fd8e-0b3f-4c87-9df8-32fb0774955d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1MDMwMTQsIm5iZiI6MTcxMTUwMjcxNCwicGF0aCI6Ii8xNDI4OTE3LzI5ODE5NTEzMS1hOTM1ZmQ4ZS0wYjNmLTRjODctOWRmOC0zMmZiMDc3NDk1NWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjdUMDEyNTE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGQzM2E3MmUwN2ZmMjhkMzFhYTJhZjlmY2I1YWUwODgxY2FhZjk0OWJmMTNhOTczMGZlNThhZTNiNzFmYzZkZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.G4aCym04gsdXwzFnL2KySDgGYLkVrlWbMykofRemFrQ) you can now copy your Brave/Vivaldi desktop file to your home, adjust perms as necessary (I didn't need to) and fingers crossed it works!

This is such a great idea @tomsjansons! I chose to try it like this, it also worked:

```bash
ln -s /run/current-system/sw/share/applications/* ~/.local/share/applications/
```

```bash
chown -hR [user]:users ~/.local/share/applications/*
```

**Comment by michaelBelsanti** [comment]
I believe this is the same issue as #160923. Try setting [xdg.portal.xdgOpenUsePortal](https://search.nixos.org/options?channel=24.11&show=xdg.portal.xdgOpenUsePortal&from=0&size=50&sort=relevance&type=packages&query=portal.).
